plugins/http_plugin/include/eosio/http_plugin/local_endpoint.hpp

```diff
@@ -0,0 +1,790 @@
+        set_tcp_pre_init_handler(h);
+    }
+
+    /// Sets the tcp post init handler
+    /**
+     * The tcp post init handler is called after the tcp connection has been
+     * established and all additional wrappers (proxy connects, TLS handshakes,
+     * etc have been performed. This is fired before any bytes are read or any
+     * WebSocket specific handshake logic has been performed.
+     *
+     * @since 0.3.0
+     *
+     * @param h The handler to call on tcp post init.
+     */
+    void set_tcp_post_init_handler(tcp_init_handler h) {
+        m_tcp_post_init_handler = h;
+    }
+
+    /// Retrieve a reference to the endpoint's io_service
+    /**
+     * The io_service may be an internal or external one. This may be used to
+     * call methods of the io_service that are not explicitly wrapped by the
+     * endpoint.
+     *
+     * This method is only valid after the endpoint has been initialized with
+     * `init_asio`. No error will be returned if it isn't.
+     *
+     * @return A reference to the endpoint's io_service
+     */
+    lib::asio::io_service & get_io_service() {
+        return *m_io_service;
+    }
+
+    /// Set up endpoint for listening manually (exception free)
+    /**
+     * Bind the internal acceptor using the specified settings. The endpoint
+     * must have been initialized by calling init_asio before listening.
+     *
+     * @param ep An endpoint to read settings from
+     * @param ec Set to indicate what error occurred, if any.
+     */
+    void listen(lib::asio::local::stream_protocol::endpoint const & ep, lib::error_code & ec)
+    {
+        if (m_state != READY) {
+            m_elog->write(log::elevel::library,
+                "asio::listen called from the wrong state");
+            using websocketpp::error::make_error_code;
+            ec = make_error_code(websocketpp::error::invalid_state);
+            return;
+        }
+
+        m_alog->write(log::alevel::devel,"asio::listen");
+
+        lib::asio::error_code bec;
+
+        {
+            boost::system::error_code test_ec;
+            lib::asio::local::stream_protocol::socket test_socket(get_io_service());
+            test_socket.connect(ep, test_ec);
+
+            //looks like a service is already running on that socket, probably another keosd, don't touch it
+            if(test_ec == boost::system::errc::success)
+               bec = boost::system::errc::make_error_code(boost::system::errc::address_in_use);
+            //socket exists but no one home, go ahead and remove it and continue on
+            else if(test_ec == boost::system::errc::connection_refused)
+               ::unlink(ep.path().c_str());
+            else if(test_ec != boost::system::errc::no_such_file_or_directory)
+               bec = test_ec;
+        }
+
+        if (!bec) {
+            m_acceptor->open(ep.protocol(),bec);
+        }
+        if (!bec) {
+            mode_t old_mask = umask(S_IXUSR|S_IXGRP|S_IRWXO);
```

Comment:
Shouldn't this honor an existing more restrictive umask?  In particular 077.

Reply:
that's a valid point

Reply:
I am thinking now, maybe should not be touching this at all